cmd/soroban-rpc/internal/methods/get_ledgers.go

```diff
@@ -0,0 +1,243 @@
+package methods
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/creachadair/jrpc2"
+
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+type LedgerPaginationOptions struct {
```

Comment:
Note for reviewers: We have the same exact pagination struct for transactions - https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/methods/get_transactions.go#L25. So, we can remove the duplication and just use a single `PaginationOptions` struct for both. However, that adds a lot of changes to this PR which are not related, so my suggestion is to do it in a separate PR.